src/hooks/useAsset/useAsset.tsx

```diff
@@ -0,0 +1,116 @@
+import { useCallback, useEffect, useMemo, useState } from 'react'
+
+import { AllChannelFieldsFragment, AssetAvailability, VideoFieldsFragment } from '@/api/queries'
+import { AssetType, useStorageProviders } from '@/hooks'
+import { createStorageNodeUrl } from '@/utils/asset'
+import { Logger } from '@/utils/logger'
+
+type UseAssetDataArgs =
+  | { entity?: VideoFieldsFragment | null; assetType: AssetType.THUMBNAIL | AssetType.MEDIA }
+  | {
+      entity?: Partial<AllChannelFieldsFragment> | null
+      assetType: AssetType.COVER | AssetType.AVATAR
+    }
+
+type UseAssetData = {
+  url?: string
+  error: boolean
+  isLoading?: boolean
+}
+
+type UseAsset = ({ entity, assetType }: UseAssetDataArgs) => UseAssetData
+
+export const useAsset: UseAsset = ({ entity, assetType }) => {
+  const { getStorageProvider } = useStorageProviders()
+  const [error, setError] = useState<boolean>(false)
+  const [isLoading, setIsLoading] = useState<boolean | undefined>(undefined)
+  const [url, setUrl] = useState<string | undefined>(undefined)
+
+  useEffect(() => {
+    if (error) {
+      Logger.error(`Fail to load ${assetType}`)
```

Comment:
I know we talked about something else but let's actually store the error (not just a boolean) and log it here as well